cni/network/network_windows.go

```diff
@@ -266,30 +266,30 @@ func getPoliciesFromRuntimeCfg(nwCfg *cni.NetworkConfig, isIPv6Enabled bool) []p
 
 		// To support hostport policy mapping
 		// uint32 NatFlagsLocalRoutedVip = 1
-		rawPolicy, _ := json.Marshal(&hnsv2.PortMappingPolicySetting{
-			ExternalPort: uint16(mapping.HostPort),
-			InternalPort: uint16(mapping.ContainerPort),
-			VIP:          mapping.HostIp,
-			Protocol:     protocol,
-			Flags:        hnsv2.NatFlagsLocalRoutedVip,
-		})
-
-		hnsv2Policy, _ := json.Marshal(&hnsv2.EndpointPolicy{
-			Type:     hnsv2.PortMapping,
-			Settings: rawPolicy,
-		})
-
-		policyv4 := policy.Policy{
-			Type: policy.EndpointPolicy,
-			Data: hnsv2Policy,
-		}
+		if net.ParseIP(mapping.HostIp).To4() != nil {
+			rawPolicy, _ := json.Marshal(&hnsv2.PortMappingPolicySetting{ // nolint
+				ExternalPort: uint16(mapping.HostPort),
+				InternalPort: uint16(mapping.ContainerPort),
+				VIP:          mapping.HostIp,
+				Protocol:     protocol,
+				Flags:        hnsv2.NatFlagsLocalRoutedVip,
+			})
 
-		logger.Info("Creating port mapping policyv4",
-			zap.Any("policy", policyv4))
-		policies = append(policies, policyv4)
+			hnsv2Policy, _ := json.Marshal(&hnsv2.EndpointPolicy{ // nolint
+				Type:     hnsv2.PortMapping,
+				Settings: rawPolicy,
+			})
+
+			policyv4 := policy.Policy{
+				Type: policy.EndpointPolicy,
+				Data: hnsv2Policy,
+			}
 
-		// add port mapping policy for v6 if we have IPV6 enabled
-		if isIPv6Enabled {
+			logger.Info("Creating port mapping policyv4",
+				zap.Any("policy", policyv4))
+			policies = append(policies, policyv4)
+		} else {
```

Comment:
There is a lot of duplication here, but the only thing that changes is the HNS flag. This could just be like
```go
hostIP, err := netip.ParseAddr(mapping.HostIp)
// handle err

var flag hnsv2.NatFlags
switch {
case hostIP.Is4():
    flag = hnsv2.NatFlagsLocalRoutedVip
case hostIP.Is6():
    flag = hnsv2.NatFlagsIPv6
}

rawPolicy := hnsv2.PortMappingPolicySetting{
    // ...
    Flags: flag,
}
```